ql/termstructures/yield/zerocurve_simple.hpp

```diff
@@ -0,0 +1,209 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2003, 2004, 2005, 2006, 2007, 2008 StatPro Italia srl
+ Copyright (C) 2009 Ferdinando Ametrano
+ Copyright (C) 2019 SoftSolutions! S.r.l.
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file zerocurve_simple.hpp
+    \brief interpolated simple zero-rates structure
+*/
+
+#ifndef quantlib_zero_curve_simple_hpp
+#define quantlib_zero_curve_simple_hpp
+
+#include <ql/math/comparison.hpp>
+#include <ql/math/interpolations/linearinterpolation.hpp>
+#include <ql/termstructures/interpolatedcurve.hpp>
+#include <ql/termstructures/yield/zeroyieldstructure_simple.hpp>
+#include <ql/utilities/dataformatters.hpp>
+#include <utility>
+
+namespace QuantLib {
+
+//! YieldTermStructure based on interpolation of zero rates
+/*! \ingroup yieldtermstructures */
+template <class Interpolator>
+class InterpolatedZeroCurveSimple : public ZeroYieldStructureSimple, protected InterpolatedCurve<Interpolator> {
```

Comment:
`InterpolatedSimpleZeroCurve`? With the file name changing accordingly?

Reply:
Yes agreed.